### DISCUSSION TITLE
Video conversion failed

### DISCUSSION THREAD

**Discussion by drmcachr** [immich-app/immich#13328] [Q&A]
[Nest] 7  - 10/10/2024, 1:16:53 AM     LOG [Microservices:MediaService] Successfully encoded 692d02cd-7115-4f50-83c5-990434315ad0
[Nest] 7  - 10/10/2024, 1:16:53 AM     LOG [Microservices:MediaService] Started encoding video f630b910-b81b-47e1-93a1-dd7e5b0b3463 {"inputOptions":["-init_hw_device qsv=hw","-filter_hw_device hw"],"outputOptions":["-c:v h264_qsv","-c:a aac","-movflags faststart","-fps_mode passthrough","-map 0:0","-strict unofficial","-map 0:1","-bf 7","-refs 5","-g 256","-v verbose","-vf format=nv12,hwupload=extra_hw_frames=64,scale_qsv=-1:720","-preset 7","-global_quality:v 23"],"twoPass":false}
[Nest] 7  - 10/10/2024, 1:17:04 AM   ERROR [Microservices:MediaRepository] ffmpeg version 6.0.1-Jellyfin Copyright (c) 2000-2023 the FFmpeg developers
  built with gcc 12 (Debian 12.2.0-14)
  configuration: --prefix=/usr/lib/jellyfin-ffmpeg --target-os=linux --extra-version=Jellyfin --disable-doc --disable-ffplay --disable-ptx-compression --disable-static --disable-libxcb --disable-sdl2 --disable-xlib --enable-lto --enable-gpl --enable-version3 --enable-shared --enable-gmp --enable-gnutls --enable-chromaprint --enable-opencl --enable-libdrm --enable-libxml2 --enable-libass --enable-libfreetype --enable-libfribidi --enable-libfontconfig --enable-libbluray --enable-libmp3lame --enable-libopus --enable-libtheora --enable-libvorbis --enable-libopenmpt --enable-libdav1d --enable-libsvtav1 --enable-libwebp --enable-libvpx --enable-libx264 --enable-libx265 --enable-libzvbi --enable-libzimg --enable-libfdk-aac --arch=amd64 --enable-libshaderc --enable-libplacebo --enable-vulkan --enable-vaapi --enable-amf --enable-libvpl --enable-ffnvcodec --enable-cuda --enable-cuda-llvm --enable-cuvid --enable-nvdec --enable-nvenc
  libavutil      58.  2.100 / 58.  2.100
  libavcodec     60.  3.100 / 60.  3.100
  libavformat    60.  3.100 / 60.  3.100
  libavdevice    60.  1.100 / 60.  1.100
  libavfilter     9.  3.100 /  9.  3.100
  libswscale      7.  1.100 /  7.  1.100
  libswresample   4. 10.100 /  4. 10.100
  libpostproc    57.  1.100 / 57.  1.100
Routing option strict to both codec and muxer layer
[AVHWDeviceContext @ 0x4d8a2100300] Trying to use DRM render node for device 0, with matching kernel driver (i915).
[AVHWDeviceContext @ 0x4d8a2100300] libva: VA-API version 1.22.0
[AVHWDeviceContext @ 0x4d8a2100300] libva: User requested driver 'iHD'
[AVHWDeviceContext @ 0x4d8a2100300] libva: Trying to open /usr/lib/jellyfin-ffmpeg/lib/dri/iHD_drv_video.so
[AVHWDeviceContext @ 0x4d8a2100300] libva: Found init function __vaDriverInit_1_22
[AVHWDeviceContext @ 0x4d8a2100300] libva: va_openDriver() returns 0
[AVHWDeviceContext @ 0x4d8a2100300] Initialised VAAPI connection: version 1.22
[AVHWDeviceContext @ 0x4d8a2100300] VAAPI driver: Intel iHD driver for Intel(R) Gen Graphics - 24.2.5 (12561f6).
[AVHWDeviceContext @ 0x4d8a2100300] Driver not found in known nonstandard list, using standard behaviour.
[AVHWDeviceContext @ 0x4d8a2100200] Use Intel(R) oneVPL to create MFX session, API version is 2.12, the required implementation version is 1.3
libva info: VA-API version 1.22.0
libva info: Trying to open /usr/lib/jellyfin-ffmpeg/lib/dri/iHD_drv_video.so
libva info: Found init function __vaDriverInit_1_22
libva info: va_openDriver() returns 0
[AVHWDeviceContext @ 0x4d8a2100200] Initialize MFX session: implementation version is 1.35
[mov,mp4,m4a,3gp,3g2,mj2 @ 0x4d8a2190900] st: 0 edit list: 1 Missing key frame while searching for timestamp: 2000
[mov,mp4,m4a,3gp,3g2,mj2 @ 0x4d8a2190900] st: 0 edit list 1 Cannot find an index entry before timestamp: 2000.
Guessed Channel Layout for Input Stream #0.1 : stereo
Input #0, mov,mp4,m4a,3gp,3g2,mj2, from '/share/Media/Videos/2024/2024-10-04｜南疆｜帕米尔之眼/C0059.MP4':
  Metadata:
    major_brand     : XAVC
    minor_version   : 17506303
    compatible_brands: XAVCmp42nrasiso6
    creation_time   : 2024-10-04T07:45:22.000000Z
  Duration: 00:00:34.56, start: 0.000000, bitrate: 217520 kb/s
  Stream #0:0[0x1](und): Video: hevc (Rext), 1 reference frame (hvc1 / 0x31637668), yuv422p10le(tv, bt709), 3840x2160 [SAR 1:1 DAR 16:9], 200300 kb/s, 50 fps, 50 tbr, 50k tbn (default)
    Metadata:
      creation_time   : 2024-10-04T07:45:22.000000Z
      handler_name    : Video Media Handler
      vendor_id       : [0][0][0][0]
      encoder         : HEVC Coding
  Stream #0:1[0x2](und): Audio: pcm_s16be (twos / 0x736F7774), 48000 Hz, 2 channels, s16, 1536 kb/s (default)
    Metadata:
      creation_time   : 2024-10-04T07:45:22.000000Z
      handler_name    : Sound Media Handler
      vendor_id       : [0][0][0][0]
  Stream #0:2[0x3](und): Data: none (rtmd / 0x646D7472), 7782 kb/s (default)
    Metadata:
      creation_time   : 2024-10-04T07:45:22.000000Z
      handler_name    : Timed Metadata Media Handler
      timecode        : 00:24:09:06
Stream mapping:
  Stream #0:0 -> #0:0 (hevc (native) -> h264 (h264_qsv))
  Stream #0:1 -> #0:1 (pcm_s16be (native) -> aac (native))
Press [q] to stop, [?] for help
[graph 0 input from stream 0:0 @ 0x4d8a21e4bc0] w:3840 h:2160 pixfmt:yuv422p10le tb:1/50000 fr:50/1 sar:1/1
[auto_scale_0 @ 0x4d8a21e4ec0] w:iw h:ih flags:'' interl:0
[Parsed_format_0 @ 0x4d8a21e4980] auto-inserting filter 'auto_scale_0' between the filter 'graph 0 input from stream 0:0' and the filter 'Parsed_format_0'
[auto_scale_0 @ 0x4d8a21e4ec0] w:3840 h:2160 fmt:yuv422p10le sar:1/1 -> w:3840 h:2160 fmt:nv12 sar:1/1 flags:0x00000004
[AVHWDeviceContext @ 0x4d8a2101700] VAAPI driver: Intel iHD driver for Intel(R) Gen Graphics - 24.2.5 (12561f6).
[AVHWDeviceContext @ 0x4d8a2101700] Driver not found in known nonstandard list, using standard behaviour.
[AVHWDeviceContext @ 0x4d8a2104200] VAAPI driver: Intel iHD driver for Intel(R) Gen Graphics - 24.2.5 (12561f6).
[AVHWDeviceContext @ 0x4d8a2104200] Driver not found in known nonstandard list, using standard behaviour.
[Parsed_scale_qsv_2 @ 0x4d8a21e4b00] Use Intel(R) oneVPL to create MFX session with the specified MFX loader
[Parsed_scale_qsv_2 @ 0x4d8a21e4b00] VPP: input is video memory surface
[Parsed_scale_qsv_2 @ 0x4d8a21e4b00] VPP: output is video memory surface
[auto_scale_0 @ 0x4d8a21e4ec0] w:3840 h:2160 fmt:yuv422p10le sar:1/1 -> w:3840 h:2160 fmt:nv12 sar:1/1 flags:0x00000004
    Last message repeated 2 times
[AVHWFramesContext @ 0x4d8a213a380] Use Intel(R) oneVPL to create MFX session, API version is 2.12, the required implementation version is 1.35
[AVHWFramesContext @ 0x4d8a213a380] Initialize MFX session: implementation version is 1.35
[h264_qsv @ 0x4d8a2202980] Using input frames context (format qsv) with h264_qsv encoder.
[h264_qsv @ 0x4d8a2202980] Encoder: input is video memory surface
[h264_qsv @ 0x4d8a2202980] Use Intel(R) oneVPL to create MFX session with the specified MFX loader
[h264_qsv @ 0x4d8a2202980] Using the intelligent constant quality (ICQ) ratecontrol method
[h264_qsv @ 0x4d8a2202980] Error initializing the encoder: invalid video parameters (-15)
[vost#0:0/h264_qsv @ 0x4d8a2211300] Error initializing output stream: Error while opening encoder for output stream #0:0 - maybe incorrect parameters such as bit_rate, rate, width or height
[AVIOContext @ 0x4d8a2160780] Statistics: 0 bytes written, 0 seeks, 0 writeouts
Terminating demuxer thread 0
[AVIOContext @ 0x4d8a21fd600] Statistics: 9987037 bytes read, 5 seeks
Conversion failed!
[Nest] 7  - 10/10/2024, 1:17:04 AM   ERROR [Microservices:MediaService] Error: ffmpeg exited with code 1: Conversion failed!
[Nest] 7  - 10/10/2024, 1:17:04 AM   ERROR [Microservices:MediaService] Error occurred during transcoding. Retrying with QSV acceleration disabled.

I got log above, some video encode error I guess, how to fix this?

**Comment by mertalev** [comment]
Below `Retrying with QSV acceleration disabled` at the end, there should be logs from it trying transcoding again. If that second attempt succeeded, you have nothing to worry about.

**Comment by drmcachr** [comment]
why this video can not convert with QSV acceleration?

**Reply by mertalev** [reply]
You might need to change "constant quality mode" in the transcoding settings to `CQP`. Older processors sometimes don't support ICQ (the default for QSV).

**Reply by drmcachr** [reply]
![image](https://github.com/user-attachments/assets/2ccc3a1c-3b24-481a-b78a-a4039061478a)

I noticed I didn't enable this settings, is that problem?

**Reply by mertalev** [reply]
I recommend enabling hardware decoding (just below the API setting) because it's much faster. But changing Auto -> CQP should be enough to fix the error (assuming this is the source of the problem).
